pkg/sql/sql_activity_update_job.go

```diff
@@ -762,3 +701,207 @@ func (u *sqlActivityUpdater) getTableRowCount(
 
 	return int64(tree.MustBeDInt(datums[0])), nil
 }
+
+// buildSelectAllTransactionsQuery is used to build the query to select and
+// aggregate all transactions for the provided aggregated timestamp to upsert
+// into the system.transaction_activity table.
+// The constructd query will expect 1 argument:
+// arg1: the aggregated timestamp to select the transactions.
+func (u *sqlActivityUpdater) buildSelectAllTransactionsQuery() string {
+	aost := u.testingKnobs.GetAOSTClause()
+	return fmt.Sprintf(`
+SELECT fingerprint_id,
+       app_name,
+       max_agg_interval,
+       metadata,
+       statistics,
+       (statistics->'statistics'->>'cnt')::int,
+       ((statistics->'statistics'->>'cnt')::float)*((statistics->'statistics'->'svcLat'->>'mean')::float),
+       COALESCE((statistics->'execution_statistics'->'contentionTime'->>'mean')::float,0),
+       COALESCE((statistics->'execution_statistics'->'cpuSQLNanos'->>'mean')::float,0),
+       (statistics->'statistics'->'svcLat'->>'mean')::float
+     FROM (
+       SELECT max(aggregated_ts) AS max_aggregated_ts,
+              app_name,
+              fingerprint_id,
+              max(agg_interval) as max_agg_interval,
+              max(metadata) as metadata,
+              merge_transaction_stats(statistics) AS statistics
+       FROM system.public.transaction_statistics %[1]s
+       WHERE aggregated_ts = $1 AND
+             app_name not like '$ internal%%'
+       GROUP BY app_name, fingerprint_id
+     ) %[1]s
```

Comment:
does an aost clause do anything on a subquery?

Reply:
It's just syntatically required for aost to work.